pkg/handlers/pvchandler.go

```diff
@@ -111,88 +124,161 @@ func (pvcHandler *PvcHandler) createPVStorage(pvc v1.PersistentVolumeClaim, pvDi
 		return
 	}
 	// set xfs_quota limit
 	subcommand := fmt.Sprintf("project -s %s", projName)
 	command := exec.Command("xfs_quota", "-x", "-c", subcommand, pvcHandler.storagePath)
 	_, err = command.CombinedOutput()
 	if err != nil {
-		log.Println("ERROR: Cannot set xfs_quota project, because: " + err.Error())
+		log.Println("PvcHandler ERROR: Cannot set xfs_quota project, because: " + err.Error())
 		return
 	}
 	subcommand = fmt.Sprintf("limit -p bhard=%s %s", storageRequest, projName)
 	command = exec.Command("xfs_quota", "-x", "-c", subcommand, pvcHandler.storagePath)
 	_, err = command.CombinedOutput()
 	if err != nil {
-		log.Println("ERROR: Cannot set xfs_quota limit, because: " + err.Error())
+		log.Println("PvcHandler ERROR: Cannot set xfs_quota limit, because: " + err.Error())
 		return
 	}
 	// Bind mounting
 	err = syscall.Mount(pvDirPath, pvDirPath, "none", syscall.MS_BIND, "")
 	if err != nil {
-		log.Println("ERROR: bind mount directories, because: " + err.Error())
+		log.Println("PvcHandler ERROR: bind mount directories, because: " + err.Error())
 		return
 	}
 	// Set fstab file
 	file, err := os.OpenFile(fstabPath, os.O_APPEND|os.O_WRONLY|os.O_SYNC, 0755)
 	if err != nil {
-		log.Println("ERROR: Cannot open fstab file: " + fstabPath + " because: " + err.Error() + "\nCannot save mountpoint!")
+		log.Println("PvcHandler ERROR: Cannot open fstab file: " + fstabPath + " because: " + err.Error() + "\nCannot save mountpoint!")
 		return
 	}
 	defer file.Close()
 	bindMountCommand := fmt.Sprintf("%[1]s %[1]s none bind 0 0\n", pvDirPath)
 	_, err = file.WriteString(bindMountCommand)
 	if err != nil {
-		log.Println("ERROR: Cannot modify fstab file: " + fstabPath + " because: " + err.Error() + "\nCannot save mountpoint!")
+		log.Println("PvcHandler ERROR: Cannot modify fstab file: " + fstabPath + " because: " + err.Error() + "\nCannot save mountpoint!")
+		return
+	}
+}
+
+func removePvDataFromFile(filePath string, searchData string) error {
+	var removedList []string
+	fileContent, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		return errors.New("Cannot read "+ filePath +" file: " + err.Error())
+	}
+	fileContentList := strings.Split(string(fileContent), "\n")
+	removeIdx := 0
+	for idx, data := range fileContentList {
+			if strings.Contains(data, searchData){
+				removeIdx = idx
+			}
+	}
+	removedList = append(removedList, fileContentList[:removeIdx]...)
+	removedList = append(removedList, fileContentList[removeIdx+1:]...)
+	file, err := os.OpenFile(filePath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0755)
+	if err != nil {
+		return errors.New("Cannot open" + filePath + " file, because: " + err.Error())
+	}
+	defer file.Close()
+	_, err = file.WriteString(strings.Join(removedList, "\n"))
+	if err != nil {
+		return errors.New("Cannot modify" + filePath + " file, because: " + err.Error())
+	}
+	return nil
+}
+
+func deletePVStorage(pv v1.PersistentVolume, storagePath string){
```

Comment:
Yeah, It was the first try. But if every delete command applied in the pvhandler, sig-storage will recreate some pvs, in case deleting many pvcs at once.
Pvhandler cannot umount fast enough, and sig-storage rediscover path and mountpoint and recreate pv.

Reply:
make PVhandler multi-threaded? :)

Reply:
but okay, dont want to torture you. a TODO is enough for now, that this should be relocated to PV Handler, and if PVs were recreated even when processing is happening in multiple threads, then it should be deleted post umount

Reply:
Thanks! :)